src/Composer/Util/GitHub.php

```
@@ -76,124 +76,125 @@ public function authorizeOAuth($originUrl)
      */
     public function authorizeOAuthInteractively($originUrl, $message = null)
     {
-        $attemptCounter = 0;
-
-        $apiUrl = ('github.com' === $originUrl) ? 'api.github.com' : $originUrl . '/api/v3';
-
         if ($message) {
             $this->io->write($message);
         }
-        $this->io->write('The credentials will be swapped for an OAuth token stored in '.$this->config->getAuthConfigSource()->getName().', your password will not be stored');
+
+        $this->io->write(sprintf('A token will be created and stored in "%s", your password will never be stored', $this->config->getAuthConfigSource()->getName()));
         $this->io->write('To revoke access to this token you can visit https://github.com/settings/applications');
+
+        $otp = null;
+        $attemptCounter = 0;
+
         while ($attemptCounter++ < 5) {
             try {
-                if (empty($otp) || !$this->io->hasAuthentication($originUrl)) {
-                    $username = $this->io->ask('Username: ');
-                    $password = $this->io->askAndHideAnswer('Password: ');
-                    $otp      = null;
+                $response = $this->createToken($originUrl, $otp);
+            } catch (TransportException $e) {
```

Alright. Should we throw an exception then, or just `write()` and return false? Or do you also want to proceed with asking for authentication credentials? Just trying to figure out how the whole flow should progress then :-)